padevents/padevents.py

```diff
@@ -603,25 +603,49 @@ async def aed_add(self, ctx, server, searchstr, group=None, time_offset: int = 0
         await self.aed_show(ctx, index)
 
     @autoeventdm.command(name="remove", aliases=['rm', 'delete'])
-    async def aed_remove(self, ctx, index: int):
+    async def aed_remove(self, ctx, index):
         """Remove an autoeventdm"""
         async with self.config.user(ctx.author).dmevents() as dmevents:
-            if not 0 < index <= len(dmevents):
-                await send_cancellation_message(ctx, "That isn't a valid index.")
-                return
-            if not await get_user_confirmation(ctx, ("Are you sure you want to delete autoeventdm with searchstring '{}'"
-                                               "").format(dmevents[index - 1]['searchstr'])):
+            if index.isdigit():
```

Comment:
Instead of duplicating this code a bunch of times, can you make a helper function called self._get_and_validate_aed_index(ctx, user_index)

you can return None from it if it's invalid, otherwise the index, and then if it's None just return

the get_and_validate function can send the cancellation message

also you can return the properly offset `index - 1` value from this function (and explain that this is what's happening in a docstring comment)